asdf/tests/conftest.py

```diff
@@ -11,3 +16,10 @@ def small_tree():
 @pytest.fixture
 def large_tree():
     return create_large_tree()
+
+
+@pytest.fixture(autouse=True)
+def restore_default_config():
+    yield
+    _config._global_config = _config.AsdfConfig()
+    _config._local = _config._ConfigLocal()
```

Comment:
Is this to avoid having to use the `with` context manager in tests?

Reply:
Yea, and global state is such a pain when it breaks tests far away.  This is a hedge that will prevent that particular frustration.

Reply:
I'm just worried that other packages will copy our test style and not realize we have an autouse fixture (and that it is needed if not using a context manager).  I'm wondering if we should eat our own dog food in our tests instead.

Reply:
I hadn't thought of that angle.  I'd prefer to keep this fixture in place, in case someone forgets to use a context manager, but I'll go ahead and update the tests with on-brand dog food :)

Reply:
(updated, and added a comment about the autouse fixture at the end of the test that sets a global config item)